EIPS/eip-1185.md

```diff
@@ -0,0 +1,80 @@
+---
+eip: 1185
+title: Storage of DNS Records in ENS
+author: Jim McDonald <Jim@mcdee.net>
+status: Draft
+type: Standards Track
+category: ERC
+created: 2018-06-26
+requires: 137
+discussions-to: https://github.com/ethereum/EIPs/pull/1185
+---
+  
+## Abstract
+This EIP defines a resolver profile for ENS that provides features for storage and lookup of DNS records. This allows ENS to be used as a store of authoritative DNS information.
+
+## Motivation
+ENS is a highly desirable store for DNS information.  It provides the distributed authority of DNS without conflating ownership and authoritative serving of information.  With ENS, the owner of a domain has full control over their own DNS records.  Also, ENS has the ability (through smart contracts) for a domain's subdomains to be irrevocably assigned to another entity.
+
+## Specification
+
+The resolver profile to support DNS on ENS builds on the standalone resolver as defined in #137.
+
+Traditionally, DNS is a zone-based system in that all of the records for a zone are kept together in the same file.  This has the benefit of simplicity and atomicity of zone updates, but when transposed to ENS can result in significant gas costs for simple changes.  As a result, the resolver works on the basis of record sets.  A record set is uniquely defined by the tuple (domain, name, resource record type), for example the tuple (example.com, www.example.com, A) defines the record set of A records for the name www.example.com in the domain example.com.  A record set can contain 0 or more values, for example if www.example.com has A records 1.2.3.4 and 5.6.7.8 then the aforementioned tuple will have two values.
+
+The choice to work at the level of record sets rather than zones means that this specification cannot completely support some features of DNS, such as zone transfers and DNSSEC.  It would be possible to build a different resolver profile that works at the zone level, however it would be very expensive to carry out updates and so is not considered further for this EIP.
+
+The DNS resolver interface consists of two functions to set DNS information and two functions to query DNS information.
+
+### setDNSRecords(bytes32 node, bytes data)
+
+`setDNSRecords()` sets, updates or clears 1 or more DNS records for a given node.  It has function signature `0x0af179d7`.
```

Comment:
Should this say "a DNS RRset", rather than "1 or more DNS records"?

Reply:
The data can contain multiple RRsets; we could change this to "1 or more DNS RRsets".